app/src/main/java/com/kms/cura/view/activity/LoginActivity.java

```diff
@@ -126,14 +127,22 @@ public void beforeTextChanged(CharSequence s, int start, int count, int after) {
            *   Disable the login button if either condition is invalid nad display an error message accordingly*/
     @Override
     public void onTextChanged(CharSequence s, int start, int before, int count) {
-        if (!InputUtils.isEmailValid(email.getText().toString())) {
-            email.setError(getResources().getString(R.string.email_error));
-            loginButton.setEnabled(false);
-
+        if (email.hasFocus()) {
+            if (!InputUtils.isEmailValid(email.getText().toString())) {
+                email.setError(getResources().getString(R.string.email_error));
+                loginButton.setEnabled(false);
+
+            } else {
+                email.setError(null);
+            }
```

Comment:
Here too. Think of some way to re-factor the common flow: if edit text has Focus -> validate -> true -> enable, false -> setError